plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -448,9 +456,17 @@ public static void verifySupportedColumnMapping(ColumnMappingMode mappingMode)
     public static List<DeltaLakeColumnMetadata> getColumnMetadata(String json, TypeManager typeManager, ColumnMappingMode mappingMode)
     {
         try {
-            return stream(OBJECT_MAPPER.readTree(json).get("fields").elements())
-                    .map(node -> mapColumn(typeManager, node, mappingMode))
-                    .collect(toImmutableList());
+            ImmutableList.Builder<DeltaLakeColumnMetadata> columns = ImmutableList.builder();
+            Iterator<JsonNode> nodes = OBJECT_MAPPER.readTree(json).get("fields").elements();
+            while (nodes.hasNext()) {
+                try {
+                    columns.add(mapColumn(typeManager, nodes.next(), mappingMode));
+                }
+                catch (UnsupportedTypeException e) {
+                    log.debug("Skip unsupported column type: %s", e.type());
```

Comment:
We can ignore on read, but can't ignore on writes, if the column has constraints. Which we won't know if we skip it completely.

What about adding a direct support for variant instead, mapping it to json?

Reply:
The write operations are protected by the unsupported writer feature `variantType`. Why are we caring about column skipping logic?

Reply:
if writes are blocked then fine

Reply:
add a code comment?

Reply:
> We can ignore on read, but can't ignore on writes

I haven't seen yet this "best effort" practice.
I find it misleading from user perspective to see only parts of the columns.

At best, such a functionality should be enabled explicitly by the user through a session property. 
This strategy is however not worth the effort.

I'm advocating either for supporting reading variants and succeed the read operation or fail the read operation otherwise.

Reply:
@findinpath Skipping unsupported column types is the default behavior in the existing connectors. 
I can't find a reason we want to change the stance only in Delta Lake connector.